# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/IonCacheIRCompiler.cpp
# Commit: 9bd1a1409856
# Full Hash: 9bd1a1409856e11810c62bd9e1e4fa57f6361818
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-07-07 09:47:47
# Regressor Bug: 1646378
# File Overlap Count: 1
# Description:
#   Bug 1646378: Add CallInlinedFunction r=jandem
#   
#   This is mostly the same as CallScriptedFunction. The key differences are:
#   
#   1. We have an extra argument, the ICScript, which must be stored in the JSContext so the Baseline prologue can find it.
# ==============================================================================

diff -r 1aa0f867fca2 -r 9bd1a1409856 js/src/jit/IonCacheIRCompiler.cpp
--- a/js/src/jit/IonCacheIRCompiler.cpp	Thu Jul 02 23:02:23 2020 +0000
+++ b/js/src/jit/IonCacheIRCompiler.cpp	Mon Jul 06 19:05:21 2020 +0000
@@ -2397,6 +2397,13 @@
   MOZ_CRASH("Call ICs not used in ion");
 }
 
+bool IonCacheIRCompiler::emitCallInlinedFunction(ObjOperandId calleeId,
+                                                 Int32OperandId argcId,
+                                                 uint32_t icScriptOffset,
+                                                 CallFlags flags) {
+  MOZ_CRASH("Call ICs not used in ion");
+}
+
 bool IonCacheIRCompiler::emitLoadArgumentFixedSlot(ValOperandId resultId,
                                                    uint8_t slotIndex) {
   MOZ_CRASH("Call ICs not used in ion");